utils.py

```diff
@@ -0,0 +1,19 @@
+from dataclasses import dataclass
+from datetime import datetime
+
+import tskit
+import tszip
+
+@dataclass
+class ARG:
```

Comment:
This seems kinda pointless, since we can define a function to compute the date of the trees we're working with (it's in sc2ts).

```python
def last_date(ts):
    if ts.num_samples == 0:
        # Special case for the initial ts which contains the
        # reference but not as a sample
        u = ts.num_nodes - 1
    else:
        u = ts.samples()[-1]
    node = ts.node(u)
    assert node.time == 0
    return parse_date(node.metadata["date"])
```

Having a date doesn't make it any more of an ARG, and this code only needs to work with sc2ts trees.

Reply:
Ah, I didn't know that was stored in the node metadata. That makes it easier, thanks.

I ripped out this class in a follow-up PR anyway, but I was still hand-hacking the start date into it.

It seems to me that the base time (corresponding to time=0 in the ts) should be in the top-level metadata, not grabbed from the last node?

Reply:
(also, I assume the last_date thing might not work on a simplified tree sequence, as the last sample might not be at time 0?)

Reply:
Yes, it would be a useful thing to have in the sc2ts top-level metadata. Pulling out from samples will work fine here though.

https://github.com/jeromekelleher/sc2ts/issues/119